src/arch/riscv/irqc/aia/interrupts.c

```diff
@@ -0,0 +1,137 @@
+/**
```

Comment:
There is a lot of logic that is replicated in the interrupts.c files. I think we should keep this logic in the riscv top-level directory interrupts.c file and for each interrupt controller (plic and aia) have an irq.c that implements the irqc-specific functionality. For example, having a generic `interrupts_arch_handle` that instead of `aplic_handle` `or plic_handle` calls `irqc_handle`, which would be an alias no one of the two (the method through which you alias is up to you, but my suggestion would be a static inline function in the irq.h header)